node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -59,7 +59,10 @@ case class NeutrinoNode(
       filterCount <- chainApi.getFilterCount
       peerMsgSender <- peerMsgSenderF
     } yield {
-      peerMsgSender.sendGetHeadersMessage(header.hashBE.flip)
+      // Get all of our cached headers in case of a reorg
+      val cachedHeaders =
```

Comment:
I don't think we want to send them all do we? Perhaps we want to sample at some exponentially increasing interval? i.e. 

current tip
2^0 
2^1 
2^2 
2^3 ....

until we run out of headers?

Reply:
We could do this but it'd complicate the logic a whole bunch, we'd have to change the `DataMessageHandler` to know what headers we have cached, as well as if we received the incorrect headers and need to send another `getheaders` message. I think this should be good enough